File: dokuz_tas/generate_sounds.py
import wave
import struct
import math
import os

os.makedirs('assets/audio', exist_ok=True)

def generate_wav(filename, duration, freq_func, vol_func, sample_rate=44100):
    num_samples = int(duration * sample_rate)
    with wave.open(filename, 'w') as wav_file:
        wav_file.setnchannels(1)
        wav_file.setsampwidth(2)
        wav_file.setframerate(sample_rate)
        
        for i in range(num_samples):
            t = float(i) / sample_rate
            freq = freq_func(t)
            vol = vol_func(t)
            
            value = int(vol * 32767.0 * math.sin(2.0 * math.pi * freq * t))
            data = struct.pack('<h', value)
            wav_file.writeframesraw(data)

# 1. Place Sound (Wooden Tok)
# Short, low frequency, quick decay
def place_freq(t): return 150.0 + (100.0 * math.exp(-t * 20))
def place_vol(t): return math.exp(-t * 30)
generate_wav('assets/audio/place.wav', 0.15, place_freq, place_vol)

# 2. Capture Sound (Sharp Break/Crunch)
# Slightly noisy, fast attack, medium decay
import random
def capture_freq(t): return 400.0 + random.uniform(-100, 100)
def capture_vol(t): return math.exp(-t * 15) * 0.8
generate_wav('assets/audio/capture.wav', 0.25, capture_freq, capture_vol)

# 3. Mill Sound (Success Ding/Shimmer)
def mill_freq(t): return 800.0 + (200.0 * math.sin(t * 20))
def mill_vol(t): return math.exp(-t * 5) * 0.5
generate_wav('assets/audio/mill.wav', 0.5, mill_freq, mill_vol)

print("Sounds generated in assets/audio/")

File: dokuz_tas/generate_sounds.dart
import 'dart:io';
import 'dart:math';
import 'dart:typed_data';

void generateWav(String filename, double duration, double Function(double) freqFunc, double Function(double) volFunc) {
  int sampleRate = 44100;
  int numSamples = (duration * sampleRate).toInt();
  
  var file = File(filename);
  file.createSync(recursive: true);
  var builder = BytesBuilder();
  
  // RIFF header
  builder.add('RIFF'.codeUnits);
  builder.add((36 + numSamples * 2).toBytes(4));
  builder.add('WAVE'.codeUnits);
  
  // fmt chunk
  builder.add('fmt '.codeUnits);
  builder.add(16.toBytes(4)); // chunk size
  builder.add(1.toBytes(2)); // PCM
  builder.add(1.toBytes(2)); // mono
  builder.add(sampleRate.toBytes(4));
  builder.add((sampleRate * 2).toBytes(4)); // byte rate
  builder.add(2.toBytes(2)); // block align
  builder.add(16.toBytes(2)); // bits per sample
  
  // data chunk
  builder.add('data'.codeUnits);
  builder.add((numSamples * 2).toBytes(4));
  
  for (int i = 0; i < numSamples; i++) {
    double t = i / sampleRate;
    double freq = freqFunc(t);
    double vol = volFunc(t);
    
    int value = (vol * 32767.0 * sin(2.0 * pi * freq * t)).toInt();
    builder.add(value.toBytes(2));
  }
  
  file.writeAsBytesSync(builder.toBytes());
}

extension IntBytes on int {
  List<int> toBytes(int length) {
    var result = <int>[];
    for (int i = 0; i < length; i++) {
      result.add((this >> (i * 8)) & 0xFF);
    }
    return result;
  }
}

void main() {
  Directory('assets/audio').createSync(recursive: true);
  
  generateWav('assets/audio/place.wav', 0.15, (t) => 150.0 + (100.0 * exp(-t * 20)), (t) => exp(-t * 30));
  generateWav('assets/audio/capture.wav', 0.25, (t) => 400.0, (t) => exp(-t * 15) * 0.8);
  generateWav('assets/audio/mill.wav', 0.5, (t) => 800.0 + (200.0 * sin(t * 20)), (t) => exp(-t * 5) * 0.5);
  
  print('Sounds generated!');
}
